### PR TITLE
[ Refactoring ] Clean driver migration

### PR DESCRIPTION
**Pull request by AlexisCnockaert**
I did a few changes based on @balsa-sarenac reviews. 
Also I changed a bit `RePushDownInstanceVariableDriver` as a choice was not working. 
Now all `gatherUserInput` methods actually fully instantiate their refactoring, as it was not done before due to old execution flow order

### CODE REVIEW

File: src/Refactoring-UI/RePullUpMethodDriver.class.st [hunk 1]
```diff
@@ -176,9 +178,9 @@ RePullUpMethodDriver >> instantiateRefactoring [
 
 	^ RePullUpMethodRefactoring
        model: model
-       pullUp: (methods collect: [ :each | each selector ])
+       pullUp: #()
        from: class name
-       to: superclass name
+       to: ''
 ]
```
Comment (balsa-sarenac) [inline src/Refactoring-UI/RePullUpMethodDriver.class.st:184]: This can work for now, but if Command passed in the selected method(s) we might be better of setting here that method(s) and it's superclass.

File: src/Refactoring-UI/ReInteractionDriver.class.st [hunk 2]
```diff
@@ -341,7 +341,8 @@ ReInteractionDriver >> runRefactoring [
 	self hasFailedApplicabilityPreconditions ifTrue: [ ^ self "applicability failed" ].
 	self hasFailedBehaviorPreservingPreconditions
 		ifTrue: [ self handleBreakingChanges ]
-		ifFalse: [ self applyChanges ]
+		ifFalse: [ self applyChanges ].
+	^ true
```
Comment (balsa-sarenac) [inline src/Refactoring-UI/ReInteractionDriver.class.st]: This is interesting, what was the reasoning behind this?
Do we use this in commands that call driver?

File: src/Refactoring-UI/RePushDownInstanceVariableDriver.class.st [hunk 3]
```diff
@@ -64,6 +64,7 @@ RePushDownInstanceVariableDriver >> gatherUserInput [
 
 	variables := self selectVariables.
 	variables ifNil: [ ^ false ].
+	self configureRefactoring.
 	^ true
```
Comment (balsa-sarenac) [inline src/Refactoring-UI/RePushDownInstanceVariableDriver.class.st:68]: I think we can avoid this If when we call setter in the refactoring
we also call the refactoring model to get the instance.
We can go like this here, but just saying it for the future, if we want to make it right everywhere

File: src/Refactoring-UI/RePushDownMethodInSomeClassesDriver.class.st [hunk 4]
```diff
@@ -43,7 +44,7 @@ RePushDownMethodInSomeClassesDriver >> instantiateRefactoring [
 		model: model 
 		pushDown: (methods collect: [ :each | each selector ])
 		from: class
-		in: classes.
+		in: #().
```
Comment (balsa-sarenac) [inline src/Refactoring-UI/RePushDownMethodInSomeClassesDriver.class.st:47]: For this, I'd say instantiate classes as empty list in the constructor, so here we have generic case

File: src/Refactoring-UI/RePushDownMethodDriver.class.st [hunk 5]
```diff
@@ -115,7 +116,7 @@ RePushDownMethodDriver >> instantiateRefactoring [
 
 	^ RePushDownMethodRefactoring
 		  model: model
-		  pushDown: (methods collect: [ :each | each selector ])
+		  pushDown: #()
```
Comment (balsa-sarenac) [inline src/Refactoring-UI/RePushDownMethodDriver.class.st:119]: unless methods is nil, I'd keep the old code here
was this failing?

File: src/Refactoring-UI/RePullUpMethodDriver.class.st (hunk 1 above)
Comment (balsa-sarenac) [reply src/Refactoring-UI/RePullUpMethodDriver.class.st:184]: same comment as below, could we keep methods list here? was it nil?

**Review by balsa-sarenac — COMMENTED** [review]
This was fast! Great work!

I've left a couple of comments below to understand some logic and a couple to do a few tweaks

File: src/Refactoring-UI/ReInteractionDriver.class.st (hunk 2 above)
Comment (AlexisCnockaert) [reply src/Refactoring-UI/ReInteractionDriver.class.st]: Yes indeed this was called in the command for Push Down iVar, it needed to return true.
Well I can keep it like that or either override it in concerned driver to do `super runRefactoring. ^true`

File: src/Refactoring-UI/RePushDownMethodDriver.class.st (hunk 5 above)
Comment (AlexisCnockaert) [reply src/Refactoring-UI/RePushDownMethodDriver.class.st:119]: it is because methods are collected with `gatherUserInput`, so passing methods here is like passing `nil` to the refactoring ( this is called before`gatherUserInput`) , leading to an error

File: src/Refactoring-UI/RePushDownInstanceVariableDriver.class.st (hunk 3 above)
Comment (AlexisCnockaert) [reply src/Refactoring-UI/RePushDownInstanceVariableDriver.class.st:68]: well instantiation was looking like 
```
ReCompositeRefactoring new
		  model: model;
		  refactorings: (variables collect: [ :v |
					   RBPushDownInstanceVariableRefactoring model: model variable: v name class: variables first definingClass ])
```
So instead of looping in and calling the setters in `gatherUserInput`, I thought this was an easier way to operate

File: src/Refactoring-UI/RePullUpMethodDriver.class.st (hunk 1 above)
Comment (AlexisCnockaert) [reply src/Refactoring-UI/RePullUpMethodDriver.class.st:184]: Same answer: it is because methods are collected with `gatherUserInput`, so passing methods here is like passing nil to the refactoring ( this is called before `gatherUserInput`) , leading to an error.

The whole flow is: first do instantiateRefactoring, but then some of the drivers are adding things to refactoring with user input. Therefore some arguments are nil from the start.

File: src/Refactoring-UI/RePushDownMethodDriver.class.st (hunk 5 above)
Comment (AlexisCnockaert) [reply src/Refactoring-UI/RePushDownMethodDriver.class.st:119]: So yes methods is nil there

File: src/Refactoring-UI/ReInteractionDriver.class.st (hunk 2 above)
Comment (balsa-sarenac) [reply src/Refactoring-UI/ReInteractionDriver.class.st]: I'd go for override to keep this clean since it's only one subclass 

but also, why does command expect true? maybe rhat is the real thing to refactor and not add return here
we can open issue if you think this is out of scope

File: src/Refactoring-UI/RePushDownInstanceVariableDriver.class.st (hunk 3 above)
Comment (balsa-sarenac) [reply src/Refactoring-UI/RePushDownInstanceVariableDriver.class.st:68]: ok, I see, this is an edge case, let's go like this, until we manage composites better

**Review by balsa-sarenac — APPROVED** [review]

**Comment by Ducasse** [comment]
looks like you have a little regression 

[unix-64 / Tests-unix-64 / Unix64.Tests-unix-64.Refactoring.UI.Tests.ReRenameInstanceVariableDriverTest.testValidNameAndUndefinedVariable](https://ci.inria.fr/pharo-ci-jenkins2/job/Test%20pending%20pull%20request%20and%20branch%20Pipeline/job/PR-19512/3/testReport/junit/Unix64.Tests-unix-64.Refactoring.UI.Tests/ReRenameInstanceVariableDriverTest/unix_64___Tests_unix_64___testValidNameAndUndefinedVariable/)
[unix-64 / Tests-unix-64 / Unix64.Tests-unix-64.Refactoring.UI.Tests.ReRenameSharedVariableDriverTest.testValidNameAndUndefinedVariable](https://ci.inria.fr/pharo-ci-jenkins2/job/Test%20pending%20pull%20request%20and%20branch%20Pipeline/job/PR-19512/3/testReport/junit/Unix64.Tests-unix-64.Refactoring.UI.Tests/ReRenameSharedVariableDriverTest/unix_64___Tests_unix_64___testValidNameAndUndefinedVariable/)

**Comment by AlexisCnockaert** [comment]
I fixed these tests

**Comment by Ducasse** [comment]
Tx!!!
